packages/story-editor/src/components/library/panes/text/karma/textPane.karma.js

```diff
@@ -122,13 +137,22 @@ describe('CUJ: Creator can Add and Write Text: Consecutive text presets', () =>
       let nodeIndex = 1;
 
       const verifyDefaultPosition = async (name, content) => {
-        await waitFor(
-          () => fixture.editor.canvas.framesLayer.frames[nodeIndex].node
-        );
+        await waitFor(() => {
+          if (!fixture.editor.canvas.framesLayer.frames[nodeIndex].node) {
+            throw new Error('node not ready');
+          }
+          expect(
+            fixture.editor.canvas.framesLayer.frames[nodeIndex].node
+          ).toBeTruthy();
+        });
         nodeIndex++;
         storyContext = await fixture.renderHook(() => useStory());
-        const element = storyContext.state.selectedElements[0];
+        let element = null;
         await waitFor(() => {
+          element = storyContext.state.selectedElements[0];
+          if (!element) {
+            throw new Error('story not ready');
+          }
```

Comment:
this is super interesting to have to do! I get that element's not quite set yet every time, but it definitely drives home why we've all scattered so many `wait waitFor WAIT PLEASE 😉 ` everywhere.